aiohttp/web_request.py

```diff
@@ -510,8 +510,26 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads):
-        """Return BODY as JSON."""
+    def json(self, *, loads=json.loads, allowed_types=None, err_handler=None):
```

Comment:
If we stay with string only argument as @socketpair suggested, that's would be friendly default behaviour with almost no cost.